### PR TITLE
PR for code review

### PR DESCRIPTION
**Pull request by antweiss**

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 name: Python package
 
-on: [pushes]
+on: [push]
```
Comment (antweiss) [inline .github/workflows/python-package.yml:6]: Very good.
A question: How do we specify this should only runs when a pull request is merged?

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -28,4 +28,8 @@ jobs:
         flake8 . --count --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest
       run: |
-        echo Add your testing here
+        pytest -rp
```
Comment (antweiss) [inline .github/workflows/python-package.yml]: Nice.
A question: how do we save test results to a file and send to Slack?

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -28,4 +28,8 @@ jobs:
...
+    - name: Publish the egg
+      run: |
+        cp .pypirc $HOME/
```
Comment (antweiss) [inline .github/workflows/python-package.yml]: Nice!
Could also be done using the [twine](https://pypi.org/project/twine/) package

File: .pypirc [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+index-servers = local
+
+[local]
+
+repository: https://otomato.jfrog.io/artifactory/api/pypi/PyPi
+
+username: candidate
+
+password: AP3dLQtysxAU27HAhoJAzUiGfYJ
```
Comment (antweiss) [inline .pypirc]: This is a security problem!
We *never* put passwords as clear unencrypted text into git. 
Please reimplement using github secrets: https://docs.github.com/en/actions/reference/encrypted-secrets

**Review by antweiss — COMMENTED** [review]
In general - very good and also impressively fast for something new to you!
See some comments in the PR

**Comment by antweiss** [comment]
@mondbev1 - there's another thing - you have 2 commits called **init1** and **init2** - while I know it's a bit of a burden - commit messages need to  be informative when working on a project with other people.

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (antweiss) [reply .github/workflows/python-package.yml]: BTW, why -rp instead of -v?

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (kuzishai) [reply .github/workflows/python-package.yml:6]: According to [events-that-trigger-workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows) I can add/change the event to pull_request (closed)

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (kuzishai) [reply .github/workflows/python-package.yml]: Echo the results to slack bot with webhooks , I might try [How to Send a Message to Slack From a Bash Script](https://www.cloudsavvyit.com/289/how-to-send-a-message-to-slack-from-a-bash-script/)

File: .pypirc (hunk 4 above)
Comment (kuzishai) [reply .pypirc]: Thank you for that.

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (antweiss) [reply .github/workflows/python-package.yml]: Yes, that's one option.
Another one is using any one of the existing Github actions for Slack:
https://github.com/marketplace/actions/github-action-for-slack#alternatives

Comment (kuzishai) [reply .github/workflows/python-package.yml]: Why -rp? It was in a manual I've found, not sure which one

Comment (antweiss) [reply .github/workflows/python-package.yml]: Gotcha!
In this case it works fine but running `pytest --help` we see:

> -r chars              show extra test summary info as specified by chars: (f)ailed, (E)rror, (s)kipped, (x)failed,
                        (X)passed, (p)assed, (P)assed with output, (a)ll except passed (p/P), or (A)ll. (w)arnings are
                        enabled by default (see --disable-warnings), 'N' can be used to reset the list. (default:
                        'fE').

Which means this only shows extra info for passed tests, not failed or skipped ones.
In CI we are usually especially interested in failed tests. So adding -rpx would be better. Or just going full cycle on `-v`

**Review by antweiss — COMMENTED** [review]
Very good!
Now please squash your commits! ;)

**Pull request closed without merging**